packages/modules/viewer-react/src/types.ts

```diff
@@ -131,22 +138,43 @@ export interface ViewerInitializerParams extends ViewerIModelAppOptions {
   /** array of iTwin.js Extensions */
   extensions?: ExtensionProvider[];
 }
+export type RequiredViewerProps = XOR<
+  XOR<ConnectedViewerProps, FileViewerProps>,
+  BlankViewerProps
+>;
+
+export type ModelLoaderProps = Partial<
```

Comment:
I think instead of doing 
```ts
Partial<ConnectedViewerProps & FileViewerProps & BlankViewerProps> & ...
```
using a union like this is more powerful:
```ts
(ConnectedViewerProps | FileViewerProps | BlankViewerProps) & ...
```

TS is able to better infer the union type, and the intellisense you get is much stronger imo
[here's an exmaple](https://www.typescriptlang.org/play?#code/C4TwDgpgBAwg9gOwRAxsCATAalAvFAbwFgAoKKAS2AHcKEAuKAZ2ACc6BzAblPIoFs4GCABtGLdgg6kAvqVKhIUAGIUREHPmJkoAMzUQACgENgAC3FtOs+SUXQAQiOMIA1psK8oAI2dvGBHIkQQrg0PD8ggh4njrGjAgArvzeEKwANF7ejN5wcOouNiShSgDyIhgxABTwSKjo2FAAPioGOC1OLu4AlFAAZLBwkYi29lAAchDUMSaswBTGIgA8tchomDgDquqbUJ1uWAB8-YPDCLYoiCxQcIzllVpe8VAAjJk62VBsiRDvfDR0RgAIiB7yKpEuCGuDAmUxi2nIzzeWUY31+XiotBhILBwSAA)

Reply:
Yes, thanks makes sense.

Reply:
Oh the issue is that we need the possibility of accepting any of the props, and the union restricts the props to one of the types:

```ts
 export type ModelLoaderProps = (
  ConnectedViewerProps | FileViewerProps | BlankViewerProps
 ) & LoaderProps;
```
![image](https://user-images.githubusercontent.com/78756012/200918571-a028cd9c-35e5-48a7-85d1-1b85868b6c53.png)

Reply:
so we cant destructure completely but would have to do props.iTwinId && props.iModelId or props.filePath instead?

Idk maybe not worth it, ill leave it up to you to resolve. thx for calling this out

Reply:
Even worse, will have to do `if ("filePath" in props)...`

Reply:
I think we can circle back with some better types once we make some breaking changes

Reply:
yea that sounds ok to me